CRM/Member/BAO/Membership.php

```diff
@@ -1835,48 +1844,20 @@ public static function renewMembership($contactID, $membershipTypeID, $is_test,
         // CRM-15475
         array_search('Cancelled', CRM_Member_PseudoConstant::membershipStatus(NULL, " name = 'Cancelled' ", 'name', FALSE, TRUE)),
       ))) {
-        $membership = new CRM_Member_DAO_Membership();
```

Comment:
Instead of bypassing membership.create call it with basic params, including the contribution